### ISSUE TITLE
Showing version number cause security concern

### ISSUE THREAD

**Issue by Haocen** [iissnan/hexo-theme-next#1873]
### I agree and want to create new issue 

- [x] Yes, I was on [Hexo Docs page](https://hexo.io/docs/), especially on [Templates](https://hexo.io/docs/templates.html), [Variables](https://hexo.io/docs/variables.html), [Helpers](https://hexo.io/docs/helpers.html) and [Troubleshooting](https://hexo.io/docs/troubleshooting.html).
- [x] Yes, I was on [NexT Documentation Site](http://theme-next.iissnan.com/).
- [x] And yes, I already searched for current [issues](https://github.com/iissnan/hexo-theme-next/issues?utf8=%E2%9C%93&q=is%3Aissue) and this is not help to me.

***

### Expected behavior 
Do not show theme version in footer.

### Actual behavior 
Showing theme version in footer.

### Steps to reproduce the behavior 
1. N/A
2. N/A
3. N/A

* Link to demo site with this issue: N/A
* Link(s) to source code or any usefull link(s): N/A

### NexT Information

**NexT Version:**

```
[x] Latest Master branch.
[] Latest Release version.
[] Old version - 
```

**NexT Scheme:**

```
[x] All schemes
[] Muse
[] Mist
[] Pisces
[] Gemini
```

### Other Information 
Although the site is static, sophisticate attack can still be used to inject malicious scripts if there is a bug in certain version of the theme.

**Comment by ivan-nginx** [comment]
In `_config.yml`:
```yml
copyright: false
```
Will not show copyright in footer, included NexT version too.

**Comment by Haocen** [comment]
@ivan-nginx This will hide the whole copyright footer.

Let me explain it in the other way:

I believe showing version number of the theme does not help either the owner of the blog, nor the visitor of the blog.

Showing the link to the theme, on the other hand, help the visitor of the blog to find this theme.

The only case one want to know the version number is the owner of the blog try to fix a problem in the theme, which can be done by looking at the json since he already has the source of the theme on the build machine already.

In conclusion, showing version number in the footer is rather useless.

**Comment by ivan-nginx** [comment]
Two ways to expand this feature:

1. Add option `version` true/false under the `copyright` option. It will be hide footer version only (json will visible in front-end and may be explained bu bots).
2. Add same option, but it will be hide both footer and json version vision.

What u think?

**Comment by Haocen** [comment]
May I ask why we want it to be visible to bots?

**Comment by ivan-nginx** [comment]
I don't want to make visible version for bots and this not actually for bots. I added version vision to fast resolve any issues, because peoples sometimes don't know wich version they use and create issues on old/wrong versions, for example.
And i'm don't thinking about bots or any security things before your issue. And u are right, need to add switch option to enable/disable version vision, but by default it must turned on like in PHP/Nginx/Apache/etc.

About JSON version — it can be used by any JS files included in `src` HexT dir, but for now it no used anywhere. I just added it on future.

Also, as a footer labels we talking here, need to add #1853 to footer copyright too and make all this things separated. Can u do this changes and create pull request?

For example, as is see this options:
```diff
-# Specify the date when the site was setup
-#since: 2015

-# Footer `powered-by` and `theme-info` copyright
-copyright: true

+footer:
+  #since: 2015
+  powered: true
+  theme: true
+  version: true

+  custom_text:
```

**Comment by Haocen** [comment]
Will take a look, doesn't seemed very complex.

Please assign the issues to me.

**Comment by ivan-nginx** [comment]
`master` branch instead of `testing` or `dev`, u may create pull right in master. Releases — are releases.

**Comment by Haocen** [comment]
```
-# icon between year and author @Footer
-authoricon: heart
```
This pending removal option does not appear to be in master, will omit instead.

**Comment by Haocen** [comment]
It is unclear to me how this option should work:

```
  # Copyright info about author or site [author | site].
  copyright: author
```

So I simply ignored it.

**Comment by ivan-nginx** [comment]
Mean here: © 2016 — 2017  **Ivan.Nginx** will be replaced by `config.author` or `config.title` or any custom text. For example, i want to show `© 2016 — 2017  AlmostOver.ru`.

**Comment by Haocen** [comment]
@ivan-nginx this option seemed to be very ambiguous?
Sometimes it is considered a string, sometimes it is considered the name of a variable.

**Comment by ivan-nginx** [comment]
And if try to parse `config.` and if it find in whole string, then this must get string from config.
And if `config.` not found in string, well, this string will be used.
Possible?

**Comment by Haocen** [comment]
How about this:

We make this option optional, if supplied, we consider it a `string` and override the `config.author` from Hexo main config.

**Comment by ivan-nginx** [comment]
Or like that, yep. Seems good too.

**Comment by ivan-nginx** [comment]
```diff
+  # Copyright info about author.
+  # If value not defined, will used author from Hexo main config.
+  copyright:
```

**Comment by Haocen** [comment]
@ivan-nginx done.
PR updated.

**Comment by ivan-nginx** [comment]
@Haocen well done! I'll check it later and make some changes (custom text, for example) and after that PR will merget. Good job and thank's for doing impact in NexT theme!

**Comment by ivan-nginx** [comment]
Merged #1886 with some improvements.
And thank u again.
